plugins/de.cau.cs.kieler.kgraph.text/src/de/cau/cs/kieler/kgraph/text/KGraphStandaloneSetup.java

```diff
@@ -25,12 +25,17 @@
  * @author msp
  */
 public class KGraphStandaloneSetup extends KGraphStandaloneSetupGenerated {
+    
+    protected static Injector injector;
 
     /**
      * Create an injector and do EMF registration.
      */
-    public static void doSetup() {
-        new KGraphStandaloneSetup().createInjectorAndDoEMFRegistration();
+    public static Injector doSetup() {
+        if (injector == null) {
+            injector = new KGraphStandaloneSetup().createInjectorAndDoEMFRegistration();
+        }
+        return injector;
```

Comment:
We keep the injector in a field to avoid that the `KGraphIdeModule`, which is registered in the `KGraphIdeSetup` is removed from the injector by someone who calls `doSetup` without ide modules in mind. I suggest to leave it this way to avoid future complications.